assets/app/scripts/directives/oscFileInput.js

```diff
@@ -31,6 +49,67 @@ angular.module('openshiftConsole')
           };
           reader.readAsText(file);
         });
+
+        // Specify event handler for the whole page drag & drop actions
+        $(document).on('dragover', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
+          $(dropArea).addClass('show-drag-and-drop-area');
+          showDropArea = true;
+        });
+        $(document).on('dragleave', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
+          showDropArea = false;
+          clearTimeout( showTimeout );
+          showTimeout = setTimeout( function(){
+              if( !showDropArea ){ $(dropArea).removeClass('show-drag-and-drop-area'); }
+          }, 200 );
```

Comment:
Can we use _.debounce or _.throttle instead of doing it ourselves with timeouts?